### PR TITLE
Fixes error states and edition set handling while creating orders

### PR DESCRIPTION
**Pull request by sweir27**
Fixes:
- removes loading state from "buy" button on error
- fixes how we assign edition set IDs
- adds a hook for `react-modal-container` for the mobile case, since it wasn't being mounted.

### CODE REVIEW

File: src/desktop/apps/artwork/components/commercial/view.coffee [hunk 1]
```diff
@@ -54,7 +54,7 @@ module.exports = class ArtworkCommercialView extends Backbone.View
     if sd.ENABLE_NEW_BUY_NOW_FLOW || loggedInUser?.hasLabFeature('New Buy Now Flow')
       serializer = new Serializer @$('form')
       data = serializer.data()
-      editionSetId: data.edition_set_id
```
Comment (ashkan18) [inline src/desktop/apps/artwork/components/commercial/view.coffee]: wait!? this is still valid js?!

File: src/desktop/apps/artwork/components/commercial/view.coffee [hunk 2]
```diff
@@ -73,9 +73,12 @@ module.exports = class ArtworkCommercialView extends Backbone.View
           if order
             location.assign("/orders/#{order.id}/shipping")
           else
+            console.error('createOrder', error)
```
Comment (ashkan18) [inline src/desktop/apps/artwork/components/commercial/view.coffee]: 👍

**Comment by ashkan18** [comment]
Merge on green

**Comment by ashkan18** [comment]
I'm pretty confused by this failed test! i checked out your branch in hope of fixing this error, everything looks fine but we get weird 
```
UnhandledPromiseRejectionWarning: Invariant Violation: Target container is not a DOM element.
```
because of new calls to `$target`. Maybe we can pair on this for fix?

**Comment by sweir27** [comment]
Yes let's pair!

File: src/desktop/apps/artwork/components/commercial/view.coffee (hunk 1 above)
Comment (sweir27) [reply src/desktop/apps/artwork/components/commercial/view.coffee]: ☹️

**Comment by mzikherman** [comment]
> What happened to yarn.lock?!

@sweir27 your `yarn` is out of date, should upgrade to 1.10.

**Comment by ashkan18** [comment]
@mzikherman should i remove merge on green?

**Comment by mzikherman** [comment]
Yea, I would say upgrade yarn and re-do the lock file. Otherwise the diff will keep going back and forth.

**Comment by sweir27** [comment]
Ah shoot, sorry. Will update my yarn.

**Comment by ashkan18** [comment]
🙌

**Pull request merged**